### PR TITLE
DR snapshot restore: use scripts provided by MCO

### PR DESCRIPTION
**Pull request by vrutkovs**
Most etcd scripts are now controlled by MCO, so `restore-cluster-state` 
function now uses those instead vendored scripts.

TODO:
* [x] Merge https://github.com/openshift/machine-config-operator/pull/775
* [x] Merge https://github.com/openshift/machine-config-operator/pull/776
* [x] Merge https://github.com/openshift/machine-config-operator/pull/779
* [x] Fix "Pod openshift-apiserver/apiserver-xxxx is not healthy: container openshift-apiserver has restarted more than 5 times" test
* [x] Fix "Prometheus when installed on the cluster should report less than two alerts in firing or pending state"
  Last node doesn't complete recovery process and it does nothing on rerun, so last node doesn't get rebooted correctly
* [x] Merge https://github.com/openshift/machine-config-operator/pull/788 - some nodes don't start as kubelet service has changed

### CODE REVIEW

**Comment by vrutkovs** [comment]
oh, that's not good:
```
Make etcd backup on first master
Creating asset directory ./assets
Downloading etcdctl binary..
etcdctl version: 3.3.10
API version: 3.3
Backing up /etc/kubernetes/manifests/etcd-member.yaml to ./assets/backup/
Error: could not open ./assets/backup/etcd/member/snap/db.part (open ./assets/backup/etcd/member/snap/db.part: no such file or directory)
```

**Comment by vrutkovs** [comment]
```
Failing tests:

[Feature:Platform] Managed cluster should have no crashlooping pods in core namespaces over two minutes [Suite:openshift/conformance/parallel]
```
Seems we need to add a two minute pause to ensure this test passes

Other test failures are flakes
```
Flaky tests:

[Feature:DeploymentConfig] deploymentconfigs keep the deployer pod invariant valid [Conformance] should deal with config change in case the deployment is still running [Suite:openshift/conformance/parallel/minimal]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity [Suite:openshift/conformance/parallel] [Suite:k8s]
```

**Comment by vrutkovs** [comment]
/hold

Waiting for MCO bugfix to land to make sure all tests pass

**Comment by vrutkovs** [comment]
```
fail [github.com/openshift/origin/test/extended/operators/cluster.go:118]: Expected
    <[]string | len:3, cap:4>: [
        "Pod openshift-apiserver/apiserver-qhcch is not healthy: container openshift-apiserver has restarted more than 5 times",
        "Pod openshift-kube-apiserver/kube-apiserver-ip-10-0-129-7.ec2.internal is not healthy: container kube-apiserver-6 has restarted more than 5 times",
        "Pod openshift-kube-apiserver/kube-apiserver-ip-10-0-140-240.ec2.internal is not healthy: container kube-apiserver-6 has restarted more than 5 times",
    ]
to be empty
```

**Comment by vrutkovs** [comment]
ssh bastion didn't start

/test pj-rehearse

**Comment by vrutkovs** [comment]
/test pj-rehearse

```
fail [github.com/openshift/origin/test/extended/operators/cluster.go:118]: Expected
    <[]string | len:2, cap:2>: [
        "Pod openshift-kube-apiserver/kube-apiserver-ip-10-0-128-107.ec2.internal is not healthy: container kube-apiserver-6 has restarted more than 5 times",
        "Pod openshift-sdn/sdn-controller-72d6f is not healthy: container sdn-controller has restarted more than 5 times",
    ]
to be empty
```

**Comment by vrutkovs** [comment]
/hold cancel

Ready for review

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 1]
```diff
@@ -236,8 +222,8 @@ objects:
           for master in "${MASTERS[@]}"
           do
             scp -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -W %h:%p core@${BASTION_HOST}" ${KUBE_SSH_KEY_PATH} "core@${master}":/home/core/.ssh/id_rsa
-            bastion_ssh "core@${master}" "chmod 0600 /home/core/.ssh/id_rsa"
-            bastion_ssh "core@${FIRST_MASTER}" "scp -o StrictHostKeyChecking=no /var/lib/etcd/snapshot.db core@${master}:/tmp/snapshot.db"
+            bastion_ssh "core@${master}" "sudo mkdir /root/.ssh; sudo -i cp /home/core/.ssh/id_rsa /root/.ssh/id_rsa && chmod 0600 /home/core/.ssh/id_rsa && sudo -i chmod 0600 /root/.ssh/id_rsa"
```
Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Why the change to copy keys to root user.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 2]
```diff
@@ -198,32 +198,18 @@ objects:
...
-          scp -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -W %h:%p core@${BASTION_HOST}" /tmp/etcd_backup.sh "core@${FIRST_MASTER}":/tmp
-          bastion_ssh "core@${FIRST_MASTER}" "sudo -i /tmp/etcd_backup.sh"
-
-          echo "Backup manifests"
-          mapfile -t MASTERS < <(oc get node -l node-role.kubernetes.io/master= -o name | cut -d '/' -f 2)
-          for master in "${MASTERS[@]}"
-          do
-            bastion_ssh "core@${master}" "sudo -i mkdir /etc/kubernetes/manifests-backup && sudo -i cp -rvf /etc/kubernetes/manifests/* /etc/kubernetes/manifests-backup"
-          done
+          bastion_ssh "core@${FIRST_MASTER}" "sudo -i /bin/bash -x /usr/local/bin/etcd-snapshot-backup.sh ./assets/backup/snapshot.db"
```
Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Also personaly I think we should absolute paths

**Comment by vrutkovs** [comment]
/cc @hexfusion @patrickdillon

Please review the sequence of actions and params to the scripts

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 2 above)
Comment (vrutkovs) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Good catch, fixed in a8f4d1346

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 1 above)
Comment (vrutkovs) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Snapshot is produced by etcdctl which is running as sudo. An easier solution is to chmod the snapshot so that `core` could access it.

Fixed in a8f4d1346

**Comment by patrickdillon** [comment]
> /cc @hexfusion @patrickdillon
> 
> Please review the sequence of actions and params to the scripts

Looks good, but I will give time for @hexfusion to take a look.

**Comment by wking** [comment]
PR topic references openshift/machine-config-operator#791.  Looks like that's been closed in favor of the still-open openshift/machine-config-operator#793?  Are we still waiting for that to land?

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 3]
```diff
@@ -182,7 +182,7 @@ presubmits:
           secretName: sentry-dsn
     trigger: '(?m)^/test (?:.*? )?e2e-aws-upgrade(?: .*?)?$'
   - agent: kubernetes
-    always_run: false
+    always_run: true
```
Comment (wking) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: I think we want a longer track-record of success before we do this (although it's really up to the MCO team).  Currently, [the past 24 hours have three failures and no success][1] for this job.

[1]: https://ci-search-ci-search-next.svc.ci.openshift.org/chart?name=e2e-restore-cluster-state

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 4]
```diff
@@ -217,7 +217,7 @@ presubmits:
           value: e2e-restore-cluster-state
         - name: TEST_COMMAND
           value: |
-            restore-cluster-state
+            restore-cluster-state
```
Comment (wking) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: nit: no need to touch this line that I can see (you're adding trailing whitespace).

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 5]
```diff
@@ -340,6 +284,10 @@ objects:
           echo "Removing ssh-bastion"
           oc delete project openshift-ssh-bastion
           oc wait namespace/openshift-ssh-bastion --for delete --timeout=10m
+
+          echo "Remove existing openshift-apiserver pods"
+          # This would ensure "Pod 'openshift-apiserver/apiserver-xxx' is not healthy: container openshift-apiserver has restarted more than 5 times" test won't fail
+          oc delete pod --all -n openshift-apiserver
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: hmm, this probably also blows away our logs for those pods?  Maybe we want to pull down their logs into the shared artifacts volume before doing this?

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 3 above)
Comment (vrutkovs) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: We can't have passing tests before MCO scripts are debugged - and we can't properly test those without having a dedicated test for DR scenarios (e.g. https://github.com/openshift/machine-config-operator/pull/793#discussion_r287329087)

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 4 above)
Comment (vrutkovs) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Rehearse test won't trigger without it

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 3 above)
Comment (wking) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: You can `/test e2e-restore-cluster-state` in that PR and it will run (and rerun after each bump) to help you debug that PR.  No need to run this in all other MCO PRs while tou debug that one.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 4 above)
Comment (wking) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: So we can drop this change once we're happy with the rehearsals?

Comment (vrutkovs) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Yeah, this would be dropped once the code is reviewed

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 3 above)
Comment (vrutkovs) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Its not clear which MCO PR would break DR scenarios. Also, if this test is misbehaving it can be skipped with `/skip` since its optional

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 4 above)
Comment (vrutkovs) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Removed this commit
